src/governance.cpp

```diff
@@ -211,13 +211,27 @@ void CGovernanceManager::ProcessMessage(CNode* pfrom, std::string& strCommand, C
 
         if(!fIsValid) {
             if(fMasternodeMissing) {
-                mapMasternodeOrphanObjects.insert(std::make_pair(nHash, object_time_pair_t(govobj, GetAdjustedTime() + GOVERNANCE_ORPHAN_EXPIRATION_TIME)));
+
+                int& count = mapMasternodeOrphanCounter[govobj.GetMasternodeVin().prevout];
+                if (count >= 10) {
+                    LogPrint("gobject", "MNGOVERNANCEOBJECT -- Too many orphan objects, missing masternode=%s\n", govobj.GetMasternodeVin().prevout.ToStringShort());
+                    // ask for this object again in 2 minutes
```

Comment:
Hmm, this isn't how AskFor is usually used. Usually it is called when the peer sends an inv.

Usually when we want to request an object we use PushMessage.

I'm not sure about the 2 minute delay. I see where that is set in AskFor but I don't see where it is used.

mapAskFor is processed by SendMessages which gets called from ThreadSocketHandler and runs much more often than every 2 minutes - 100 ms, I think.